### PR TITLE
Copy method

### PR DESCRIPTION
**Pull request by yanamal**
Hi David,
Here is the copy method branch, but it is still unfinished:
the turtles are not copying the shell, and there is some ugliness with having to manually copy some properties.

### CODE REVIEW

**Comment by davidbau** [comment]
Be sure to add your new "copy" function to the list of expected function names in the test that failed.  The test is designed to make sure that we don't leak any _additional_ names that are unexpected.

**Comment by davidbau** [comment]
Clean code snippets from stack overflow -

You can recopy all attributes using this:
http://stackoverflow.com/questions/6753362/jquery-how-to-copy-all-the-attributes-of-one-element-and-apply-them-to-another

And clone the canvas:
http://stackoverflow.com/questions/3318565/any-way-to-clone-html5-canvas-element-with-its-content

**Comment by yanamal** [comment]
I think it's ready now!

File: jquery-turtle.js [hunk 1]
```diff
@@ -6444,6 +6444,50 @@ var turtlefn = {
...
+      
+      // copy attributes, just in case:
+      var attrs = this.prop("attributes");
+      //console.log(attrs)
+      for(i in attrs) {
+        t2.attr(attrs[i].name, attrs[i].value);
+      }
+      
+      // copy the canvas:
+      t2.canvas().width = this.canvas().width;
```
Comment (yanamal) [inline jquery-turtle.js]: David says: check if canvas returns anything before using it...

Also, consider adding a test with a non-canvas turtle.

**Comment by davidbau** [comment]
Thanks yana!

**Pull request merged**